deep_learning_model.py

```
@@ -0,0 +1,81 @@
+def prepare_features(df: pd.DataFrame) -> pd.DataFrame:
+    features = df.copy()
+    features['yards_per_attempt'] = features['rushing_yards'] / (features['rushing_attempts'] + 1)
+    features['receptions_per_target'] = features['receptions'] / (features['targets'] + 1)
+    features['position_encoded'] = features['position'].map({'RB':1,'WR':2,'TE':3,'QB':4}).fillna(0)
+
+    lag_cols = ['rushing_yards','receiving_yards','rushing_attempts','receptions']
+    features = features.sort_values(['player_id','season']).reset_index(drop=True)
+    for col in lag_cols:
+        features[f'{col}_prev'] = features.groupby('player_id')[col].shift(1).fillna(0)
+    for col in lag_cols:
+        career_avg = features.groupby('player_id')[col].expanding().mean().reset_index(level=0,drop=True)
```

Pylint (3.3.7)

[convention] Line too long (105/100)

(C0301)